lib/ethpy/ethpy/base/retry_utils.py

```diff
@@ -55,27 +70,29 @@ async def async_retry_call(
             caller = inspect.stack()[1][3]
             logging.warning(
                 "Retry attempt %s out of %s: Function %s called from %s failed with %s",
-                attempt_number,
+                attempt_number + 1,
                 retry_count,
                 func,
                 caller,
                 repr(exc),
             )
             exception = exc
-            # TODO implement smarter wait here
-            await asyncio.sleep(0.1)
+            await asyncio.sleep(start_latency * backoff_multiplier**attempt_number)
     assert exception is not None
     raise exception
 
 
 def retry_call(
     retry_count: int,
     retry_exception_check: Callable[[Exception], bool] | None,
-    func: Callable[P, R],
-    *args: P.args,
-    **kwargs: P.kwargs,
+    func: Callable[..., R],
```

Comment:
I would take the time to organize these better.  func, *args and **kwargs should come last.  Put the options first with the other options.  Optionally we could make an options object to hold all the options:
```python
def retry_call(
    options: RetryCallOptions,
    func: Callable[P, R],
    *args: P.args,
     **kwargs: P.kwargs
```

Reply:
ah! adding `  # type: ignore` stops pyright from complaining about the latter!

Reply:
removed the part